# SPDX-FileCopyrightText: 2022 fosslinux <fosslinux@aussies.space>
#
# SPDX-License-Identifier: curl

Regenerating help is not trivial. Help is unnecessary.
Disable help.

diff -ru src/Makefile.inc src/Makefile.inc
--- curl-8.5.0/src/Makefile.inc
+++ curl-8.5.0/src/Makefile.inc
@@ -76,12 +76,10 @@
   tool_formparse.c \
   tool_getparam.c \
   tool_getpass.c \
-  tool_help.c \
   tool_helpers.c \
   tool_hugehelp.c \
   tool_ipfs.c \
   tool_libinfo.c \
-  tool_listhelp.c \
   tool_main.c \
   tool_msgs.c \
   tool_operate.c \
@@ -120,7 +118,6 @@
   tool_formparse.h \
   tool_getparam.h \
   tool_getpass.h \
-  tool_help.h \
   tool_helpers.h \
   tool_hugehelp.h \
   tool_ipfs.h \
diff -ru src/tool_operate.c src/tool_operate.c
--- curl-8.5.0/src/tool_operate.c
+++ curl-8.5.0/src/tool_operate.c
@@ -78,7 +78,6 @@
 #include "tool_writeout.h"
 #include "tool_xattr.h"
 #include "tool_vms.h"
-#include "tool_help.h"
 #include "tool_hugehelp.h"
 #include "tool_progress.h"
 #include "tool_ipfs.h"
@@ -2746,19 +2745,7 @@
     if(res) {
       result = CURLE_OK;
 
-      /* Check if we were asked for the help */
-      if(res == PARAM_HELP_REQUESTED)
-        tool_help(global->help_category);
-      /* Check if we were asked for the manual */
-      else if(res == PARAM_MANUAL_REQUESTED)
-        hugehelp();
-      /* Check if we were asked for the version information */
-      else if(res == PARAM_VERSION_INFO_REQUESTED)
-        tool_version_info();
-      /* Check if we were asked to list the SSL engines */
-      else if(res == PARAM_ENGINES_REQUESTED)
-        tool_list_engines();
-      else if(res == PARAM_LIBCURL_UNSUPPORTED_PROTOCOL)
+      if(res == PARAM_LIBCURL_UNSUPPORTED_PROTOCOL)
         result = CURLE_UNSUPPORTED_PROTOCOL;
       else if(res == PARAM_READ_ERROR)
         result = CURLE_READ_ERROR;
